src/value_generators/test/DocumentGeneratorTestCases.yml

```diff
@@ -592,3 +592,18 @@ Tests:
     ThenReturns:
     - { "date" : { "$date" : { "$numberLong" : "1369875424299" } } }
 
+  - Name: SimpleIncrement
+    GivenTemplate:
+      int: {^Inc: {start: 1, multiplier: 100, step: 1}}
+    ThenReturns:
+    - "int": { "$numberLong" : "201"}
+    - "int": { "$numberLong" : "202"}
+    - "int": { "$numberLong" : "203"}
+
+  - Name: IncrementWithStep
+    GivenTemplate:
+      int: {^Inc: {start: 100, multiplier: 200, step: 2}}
+    ThenReturns:
+    - "int" : { "$numberLong" :"500"}
+    - "int" : { "$numberLong" :"502"}
+    - "int" : { "$numberLong" :"504"}
```

Comment:
I think there is one or two other test-cases that are commented out because of that bson bug and would hopefully work now - can you try uncommenting them?

(Unfortunately this yaml-based testing infrastructure relies on the RNG returning values in the same order, so adding test-cases in the middle of the file involves changing all the expected values that occur after those changes. If this is too tedious lemme know I can pull your branch and help with that part.)

Reply:
I was going to ask the same thing. 

Question for Alex and Ryan. Should we reseed the RNG between test cases? That would allow us to call it multiple times to get different values, but we wouldn't carry state from one test to the next. It would probably require updating all of the expected values though :(

Reply:
> Should we reseed the RNG between test cases?

[PERF-2118](https://jira.mongodb.org/browse/PERF-2118) is to do this, but it's actually only a 2-line change - just move this line:

https://github.com/mongodb/genny/blob/master/src/value_generators/test/DocGenYamlTestCaseRunner.cpp#L22

down to the top of the `DYNAMIC SECTION`:

https://github.com/mongodb/genny/blob/master/src/value_generators/test/DocGenYamlTestCaseRunner.cpp#L68

This will of course break all the existing `ThenReturns` values though, as David noted.